cs/src/core/Index/Recovery/Checkpoint.cs

```diff
@@ -673,7 +497,7 @@ private SystemState GetNextState(SystemState start, CheckpointType type = Checkp
                     nextState.version = start.version + 1;
                     break;
                 case Phase.IN_PROGRESS:
-                    nextState.phase = Phase.WAIT_PENDING;
+                    nextState.phase = RelaxedCPR ? Phase.WAIT_FLUSH : Phase.WAIT_PENDING;
```

Comment:
Please update the comment above the `GetNextState` and `GlobalMoveToNextState` functions to reflect the new state machine.